Automattic/themes
# Blockbase: Remove style variations

#### Changes proposed in this Pull Request:

As of [Gutenberg 15.1](https://github.com/WordPress/gutenberg/pull/46554)/[WordPress 6.2](https://github.com/WordPress/wordpress-develop/pull/3901), child themes inherit the style variations of the parent theme.

This is causing issues to Blockbase child themes such as Jackson in WordPress.com, because the headstart content for a child theme is only created for the default style variation. When a different style variation is applied, the styles from the headstart content are incompatible with the new variation:

https://user-images.githubusercontent.com/1233880/230398566-f5abd9a1-c558-4016-9734-201c430d81d3.mov


To mitigate that, and following [this suggestion](https://github.com/Automattic/themes/pull/6996#issuecomment-1500275877) from @pbking, this PR removes the style variations from the Blockbase theme, so users are no longer able to select them for either the parent or any child theme.

This is an alternate approach of https://github.com/Automattic/themes/pull/6996 / D107265-code, which only removes the variations for Blockbase child themes and keeps them on the parent theme.

#### Testing instructions:

- Apply these changes to your WP.com sandbox: `npm run deploy:push:changes`
- Also apply these changes to your sandbox to avoid issues with the `themes` cache while testing:

```diff
diff --git a/wp-content/lib/wpcom-themes.php b/wp-content/lib/wpcom-themes.php
index 534de104e0..7a2e46d7d9 100644
--- a/wp-content/lib/wpcom-themes.php
+++ b/wp-content/lib/wpcom-themes.php
@@ -2099,7 +2099,8 @@ class WPCom_Themes {
        }
 
        public static function get_style_variations( $stylesheet ) {
-               $style_variations = wp_cache_get( "style_variations_$stylesheet", 'themes' );
+               //$style_variations = wp_cache_get( "style_variations_$stylesheet", 'themes' );
+               $style_variations = false;
 
                if ( $style_variations === false ) {
                        $theme_demo_site = wpcom_get_theme_demo_site( $stylesheet );
@@ -2156,7 +2157,7 @@ class WPCom_Themes {
                                $style_variation['slug'] = sanitize_title( $style_variation['title'] );
                        }
 
-                       wp_cache_set( "style_variations_$stylesheet", $style_variations, 'themes', 5 * MINUTE_IN_SECONDS );
+                       //wp_cache_set( "style_variations_$stylesheet", $style_variations, 'themes', 5 * MINUTE_IN_SECONDS );
                }
 
                return $style_variations;
```
- Sandbox the API
- It's also recommended to use an incognito window/private session to avoid issues with the cached API responses by Calypso.
- Go to the WP.com Design Picker (`wordpress.com/setup/site-setup/designSetup?siteSlug=<SITE_DOMAIN>`)
- Search a Blockbase child theme such as Jackson
- Make sure it does not have any style variation
- Go to WP.com > Appearance > Themes
- Search a Blockbase child theme such as Jackson
- Make sure it does not have any style variation
- Activate it
- Go to WP.com > Appearance > Editor
- Open the Styles sidebar
- Make sure it does not have any style variation

#### Related issue(s):
Fixes https://github.com/Automattic/themes/issues/6941.
